### PR TITLE
refactor: Functional SSO components

### PR DESCRIPTION
**Pull request by Yserz**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "raygun4js": "2.18.0",
     "react": "16.10.2",
     "react-dom": "16.10.2",
-    "react-intl": "2.9.0",
+    "react-intl": "3.3.2",
```
Comment (bennycode) [inline package.json:34]: When updating `react-intl` we should also update `@types/react-intl`.

File: src/script/auth/page/SingleSignOn.tsx [hunk 2]
```diff
@@ -32,31 +32,32 @@ import {
...
 import Page from './Page';
 import SingleSignOnForm from './SingleSignOnForm';
 
-interface Props extends React.HTMLAttributes<SingleSignOn>, RouteComponentProps<{}> {}
+interface Props extends React.HTMLAttributes<HTMLDivElement> {}
 
-interface State {
-  isOverlayOpen: boolean;
-}
+let ssoWindow: Window = undefined;
```
Comment (AndyLnd) [inline src/script/auth/page/SingleSignOn.tsx]: useRef

File: src/script/auth/page/SingleSignOn.tsx [hunk 3]
```diff
@@ -176,82 +177,93 @@ class SingleSignOn extends React.PureComponent<Props & InjectedIntlProps, State>
     const top = parentHeight / 2 - childHeight / 2 + screenTop;
     return {left, top};
   };
+  const focusChildWindow = () => ssoWindow && ssoWindow.focus();
```
Comment (AndyLnd) [inline src/script/auth/page/SingleSignOn.tsx]: focusSsoWindow

File: package.json (hunk 1 above)
Comment (Yserz) [reply package.json:34]: We can't do that until we moved all components to functional components. Reason for that is that react-intl moved the types to the package itself and the 3.x types are not working anymore.

Comment (Yserz) [reply package.json:34]: Also the temporary solution for using the hook won't work.

**Pull request merged**